src/room_member.cpp

```diff
@@ -28,6 +28,24 @@ RoomMember::~RoomMember() {
     RakNet::RakPeerInterface::DestroyInstance(peer);
 }
 
+void RoomMember::HandleChatPacket(const RakNet::Packet* packet) {
+    RakNet::BitStream stream(packet->data, packet->length, false);
+
+    // Ignore the first byte, which is the message id.
+    stream.IgnoreBytes(sizeof(RakNet::MessageID));
+
+    ChatEntry chat_entry{};
+    RakNet::RakString nickname;
+    stream.Read(nickname);
+    chat_entry.nickname.assign(nickname.C_String(), nickname.GetLength());
+    RakNet::RakString message;
+    stream.Read(message);
+    chat_entry.message.assign(message.C_String(),message.GetLength());
+
+    std::lock_guard<std::mutex> lock(chat_mutex);
+    chat_queue.emplace_back(std::move(chat_entry));
```

Comment:
please put the lock_guards in a seperate scope so it's very clear where it starts and where it ends.

Next time someone adds a sleep below this line and doesn't realize what they caused.